vignettes/GenomicFiles.Rmd

```diff
@@ -0,0 +1,727 @@
+`elementMetadata` and `assays` slots.
+
+```{r overview-GenomicFiles}
+GenomicFiles(gr, fls)
+```
+
+A `GenomicFiles` can be used as the `ranges` argument to the functions in this
+package. When `summarize=TRUE`, data from the common slots are transferred to
+the `SummarizedExperiment` result. NOTE: Results can only be put into a
+`SummarizedExperiment` when no reduction is performed because of the matching
+dimensions requirement (i.e., a REDUCE collapses the results in one dimension).
+
+## Functions
+
+Functions in `GenomicFiles` manipulate and combine data across or within files
+using the parallel infrastructure provided in `BiocParallel`. Files and ranges
+are sent to workers along with MAP and REDUCE functions. The MAP extracts and/or
+manipulates data and REDUCE consolidates the results from MAP. Both MAP and
+REDUCE are executed in the distributed step and therefore reduction occurs on
+data from the same worker, not across workers. The chart in Figure
+[1](reduceByRange_flow.png) represents the division of labor in `reduceByRange`
+and `reduceRanges` with 3 files and 4 ranges. These functions split the problem
+by range which allows subsets (i.e., the same range) to be combined across
+different files. `reduceByRange` iterates through the files, invoking MAP and
+REDUCE for each range / file combination. This approach allows ranges extracted
+from the files to be kept separate or combined before the next call to `MAP`
+based on whether or not a `REDUCE` is supplied.
+
+`reduceRanges` applies `MAP` to each range / file combination and REDUCEs the
+output of all MAP calls. `REDUCE` usually plays a minor role by concatenating or
+unlisting results.
+
+```{r echo = FALSE, fig1, fig.cap=": Mechanics of `reduceByRange` and `reduceRanges`"}
+knitr::include_graphics("reduceByRange_flow.png")
+```
+
+In contrast to the 'byRange' approach, `reduceByFile` and `reduceFiles` (Figure
+[2](reduceByFile_flow.png)) split the problem by file. Files are sent to
+different workers with the set of ranges allowing subsets (i.e., multiple
+ranges) from the same file to be combined. `reduceByFile` invokes MAP for each
+file / range combination allowing potential `REDUCE` after each MAP step.
+
+`reduceFiles` applies `MAP` to each range / file combination and REDUCEs the
+output of all MAP calls. `REDUCE` usually plays a minor role by concatenating or
+unlisting results.
+
+```{r echo = FALSE, fig2, fig.cap=": Mechanics of `reduceByFile` and `reduceFiles`"}
+knitr::include_graphics("reduceByFile_flow.png")
+```
+
+# Queries across files: `reduceByRange` and `reduceRanges`
+
+The `reduceByRange` and `reduceRanges` functions are designed for analyses that
+compare or combine data subsets across files. The first example in this section
+computes pileups on subsets from individual files then sums over all files. The
+second example computes coverage on a group of ranges for each file then
+performs a basepair-level t-test across files. The t-test example also
```

Comment:
When I look at `t-test` in the original, the `t` looks like it has special formatting. Please change all instances of `t-test` to be formatted as

    `t`-test